Konflux: 1.5: ART updates

rayfordj opened the pull request:
Changes:
 - update mustgather pullspec
 - imagePullPolicy: IfNotPresent
 - update image-references
 - move image-references

rayfordj committed:
Konflux: 1.5: ART updates

Changes:
 - update mustgather pullspec
 - imagePullPolicy: IfNotPresent
 - update image-references
 - move image-references

weshayutin reviewed (dismissed):
where do the git commits that are pinned for a release live now?

rayfordj commented:
/hold

rayfordj committed:
Update 'make bundle' to skip image-references

```
$ BRANCH_VERSION="oadp-1.5" make bundle
   ...
ERRO[0001] Error: Value image.openshift.io/v1, Kind=ImageStream:
unsupported media type registry+v1 for bundle object

make: *** [Makefile:339: bundle] Error 1

$ yq .apiVersion bundle/manifests/image-references
"image.openshift.io/v1"
```

rayfordj commented:
/unhold

rayfordj commented:
/retest-required

rayfordj commented:
/retest-required

rayfordj commented:
/retest-required

rayfordj commented:
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_oadp-operator/1962/pull-ci-openshift-oadp-operator-oadp-1.5-4.19-e2e-test-aws/1969771295990091776
```
{  could not initialize namespace: failed to wait for authentication cache to warm up after 15s: 
projects.project.openshift.io "ci-op-18s0xptm" is forbidden: User "system:serviceaccount:ci:ci-operator" 
cannot get resource "projects" in API group "project.openshift.io" in the namespace "ci-op-18s0xptm"}
```

/retest-required

weshayutin commented:
if we can get the 4.19 jobs to pass here, the 4.20 jobs can be skipped

rayfordj committed:
Revert "Update 'make bundle' to skip image-references"

This reverts commit b9000d95f63ae28e1903952b75722a54e33bc448.

rayfordj committed:
Move bundle/manifests/image-references

```
bundle/manifests/image-references -> bundle/image-references
```

[Slack Thread1](https://redhat-internal.slack.com/archives/C039LRSDC8Z/p1758556848172319?thread_ts=1758292522.363369&cid=C039LRSDC8Z)
[Slack Thread2](https://redhat-internal.slack.com/archives/C039LRSDC8Z/p1758562213259309)

weshayutin reviewed (approved):
/approved /LGTM

shubham-pampattiwar reviewed (approved):
/lgtm
